### DISCUSSION TITLE
Help me prevent file uploads with custom validation rules

### DISCUSSION THREAD

**Discussion by tanthammar** [livewire/livewire#1400] [4. General]
I am trying to hook into file upload validation errors. 

I have set main rules in the `config/livewire.php`  but I want to prevent files from being uploaded when I define custom validation in certain forms.  If I use the `updatedFoo()` method to validate, it is to late. The file is already uploaded.

```php
public function updatedPhoto()
    {
        $this->validate([
            'photo' => 'image|max:1024', // this is to late, the file is already uploaded
        ]);
    }
```

**Comment by tanthammar** [comment]
What I want to achieve is something like this
```php
     public function updatedPhotos()
    {
        try {
            $this->validateOnly('photos.*', 'image|max:1024');
        } catch (ValidationException $e) {
            $this->deleteAllTempFiles('photos');
        }
    }
```

**Comment by tanthammar** [comment]
I tried a crazy work around because I was unable to catch the validation error event in backend.

```blade
@error('photos.*')
   @php $this->deleteAllTempFiles('photos') @endphp
@enderror 
```
Found that this was an **extremely bad idea** :) to call a method, that sets a component property, if the form has errors. It causes a loop with ajax requests.)

(plus I ran into a "bug", `$this` looses its context depending on where the input is positioned in the form. )
#1401

**Comment by colynb** [comment]
@tanthammar an luck with this? I'm new to Livewire and I am having a similar issue. Just from experimenting, I found a way to hook into the change event of the file input:

```
<input type="file" name="upload" wire:model="video" class="hidden" accept="video/*"
                            wire:change="$emitSelf('fileChanged', {type: $event.target.files[0].type, size: $event.target.files[0].size})" />
```

I pass in the size and type of the file to the component even listener so I can validate them in PHP... trouble is I don't know how to tell Livewire to cancel the upload if needed. I can throw a validation error but the upload continues.

**Comment by usernotnull** [comment]
+1 I'm actually looking for a way to cancel an ongoing upload and a solution for this would also work for my case.
